packages/core/src/server.ts

```diff
@@ -582,6 +598,20 @@ export class Controller extends API {
         // authentication.
         if (!(await getCryptoProvider().timingSafeEqual(M, srp.M1!))) {
             this.log("account.createSession", { success: false });
+            ++srpState.failedAttempts;
+            await this.storage.save(auth);
+            if (srpState.failedAttempts >= 5 && acc.settings.notifications.failedLoginAttempts) {
+                try {
+                    const location = this._buildLocationAndDeviceString(this.context.location, this.context.device);
+
+                    await this.messenger.send(acc.email, new FailedLoginAttemptMessage({ location }));
+
+                    // Remove trusted device, if it was (after email as this can throw if the device was already removed)
+                    if (this.context.device) {
+                        await this.removeTrustedDevice(this.context.device.id);
+                    }
```

Comment:
This should be outside of the conditional block, since we should probably do this regardless of the notification setting. We can also add a separate setting for this but I think simply doing this as a default security precaution is totally justifiable and doesn't necessarily need to be disable-able.